src/Serializer/AbstractItemNormalizer.php

```diff
@@ -481,4 +491,34 @@ protected function normalizeRelation(PropertyMetadata $propertyMetadata, $relate
 
         return $iri;
     }
+
+    /**
+     * Populates the object if allowed to and not set already. If you pass identifiers, it will search for them, otherwise
+     * fall back to the identifiersextractor.
+     *
+     * @param $data
+     * @param $class
+     * @param array $context
+     * @param array $identifiers
+     */
+    protected function populateObjectIfAllowed($data, $class, array $context = [], array $identifiers = [])
+    {
+        if (!isset($context[self::OBJECT_TO_POPULATE])) {
+            if (0 === count($identifiers)) {
+                $identifiers = $this->identifiersExtractor->getIdentifiersFromResourceClass($class);
+            }
+
+            if (0 !== count(array_intersect(array_keys($data), $identifiers))) {
```

Comment:
I have no clue if this works, it was just a quick hack and needs to be covered by tests. The idea here is just to check if the data contains all identifiers.